web/directives/toast-notifications.js

```diff
@@ -98,6 +98,10 @@ module.exports = function ($timeout) {
                 let normalEvent = (title) => {
                     return (event) => {
                         let data = JSON.parse(event.data);
+
+                        const customEvent = new CustomEvent('PushEvent', { detail: data.detail });
+                        window.dispatchEvent(customEvent);
```

Comment:
What's this supposed to do? It seems unrelated to plugins

Reply:
This event is a frontend only event, this sends push's data to all `EventListener` window on channel `PushEvent`.

With this custom event, all web plugins can receive push information on a isolate js scope and can communicate with another plugins without the backend, it's a "modern version" of postMessage 🤢, but better, lighter, for current window scope (not for wall browser windows) and recommended instead of postMessage.

For now, we use this only for `plugin.uninstall` and `plugin.install` events sent from backend over EventSource. With this events on `web/app.js` we reload page and store restart information on web scope to show restart alert.

👍🏻